improver/__init__.py

```diff
@@ -33,6 +33,7 @@
 from abc import ABC, abstractmethod
 from collections.abc import Iterable
 
+from dagrunner import Plugin
```

Comment:
I think I will be removing dependence on the dagrunner.
The intention is that dagrunner will document a really basic UI that it will support but not impose that you subclass from it.
```suggestion
```

Reply:
I have removed dependence on dagrunner for this abstract plugin class for now.